src/mixins/filing-template-mixin.ts

```diff
@@ -312,6 +329,51 @@ export default class FilingTemplateMixin extends Vue {
     // Will refactor above parse function into 1 generic filing parse function when we know more
   }
 
+  /**
+   * Parses a business snapshot into the store.
+   * @param businessSnapshot the business to be parsed
+   */
+  parseBusinessSnapshot (businessSnapshot: BusinessSnapshotIF[]): void {
+    if (businessSnapshot.length !== 6) throw new Error('Incomplete request responses  \'businessIdentifier\'')
+
+    // Set Business Information
+    this.setBusinessInformation(businessSnapshot[0].business)
+
+    // Set Name Translations
+    this.setNameTranslations(
+      businessSnapshot[1].aliases?.map(x => {
+        return {
+          value: x.alias,
+          oldValue: null,
+          action: null
+        }
+      })
+    )
+
+    // Set Office Addresses
+    this.setOfficeAddresses(businessSnapshot[2])
+
+    // Set People and Roles
+    this.setPeopleAndRoles(businessSnapshot[3].directors?.map(director => {
+      return {
+        ...director,
+        roles: [
+          {
+            roleType: director.role,
+            appointmentDate: director.appointmentDate,
+            cessationDate: null
+          }
+        ]
```

Comment:
Pretty fancy!

I would prefer that you not spread `director` above, as this will result in invalid properties in the director record (OrgPersonIF) -- specifically, appointmentDate, cessationDate and role.

Reply:
k